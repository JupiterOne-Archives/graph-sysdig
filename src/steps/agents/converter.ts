import {
  createIntegrationEntity,
  Entity,
} from '@jupiterone/integration-sdk-core';

import { SysdigAgent } from '../../types';
import { Entities } from '../constants';

export function createAgentEntity(data: SysdigAgent): Entity {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _key: `sysdig-agent:${data.clusterName}-${data.machineId}-${data.labels.provider}-${data.labels.hostname}`,
        _type: Entities.AGENT._type,
        _class: Entities.AGENT._class,
        name: data.labels.hostname,
        agentStatus: data.agentStatus,
        clusterName: data.clusterName,
        accountId: data.labels.accountId,
        hostname: data.labels.hostname,
        machineId: data.labels.machineId,
        provider: data.labels.provider,
        agentConnectString: data.moreOptions.agentConnectString,
        category: ['application'],
      },
    },
  });
}
