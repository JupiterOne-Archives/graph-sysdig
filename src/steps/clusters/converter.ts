import {
  createIntegrationEntity,
  Entity,
  parseTimePropertyValue,
} from '@jupiterone/integration-sdk-core';

import { SysdigCluster } from '../../types';
import { Entities } from '../constants';

export function getClusterKey(provider: string, name: string) {
  return `sysdig-cluster:${provider}-${name}`;
}

export function createClusterEntity(data: SysdigCluster): Entity {
  const { agentConnectString, ...cluster } = data;

  return createIntegrationEntity({
    entityData: {
      source: {
        ...cluster,
      },
      assign: {
        _key: getClusterKey(data.provider, data.name),
        _type: Entities.CLUSTER._type,
        _class: Entities.CLUSTER._class,
        customerId: data.customerID,
        accountId: data.accountID,
        provider: data.provider,
        name: data.name,
        region: data.region,
        zone: data.zone,
        agentConnected: data.agentConnected,
        createdOn: parseTimePropertyValue(data.createdAt),
        nodeCount: data.nodeCount,
        clusterResourceGroup: data.clusterResourceGroup,
        version: data.version,
      },
    },
  });
}
