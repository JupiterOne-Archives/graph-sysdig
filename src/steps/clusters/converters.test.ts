import { getMockCluster } from '../../../test/mocks';
import { createClusterEntity } from './converter';

describe('#createClusterEntity', () => {
  test('should convert to entity', () => {
    expect(createClusterEntity(getMockCluster())).toMatchSnapshot();
  });
});
