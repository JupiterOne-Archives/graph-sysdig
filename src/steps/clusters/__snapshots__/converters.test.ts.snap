// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createClusterEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Cluster",
  ],
  "_key": "sysdig-cluster:gcp-sysdig-cluster",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "accountID": "30521231234",
        "agentConnected": false,
        "clusterResourceGroup": "",
        "createdAt": "2002-10-22T09:22:26Z",
        "customerID": 123456,
        "name": "sysdig-cluster",
        "nodeCount": 1,
        "provider": "gcp",
        "region": "us-central1",
        "version": "1.23.8-gke.1900",
        "zone": "us-central1-c",
      },
    },
  ],
  "_type": "sysdig_cluster",
  "accountId": "30521231234",
  "agentConnected": false,
  "clusterResourceGroup": "",
  "createdOn": 1035278546000,
  "customerId": 123456,
  "displayName": "sysdig-cluster",
  "name": "sysdig-cluster",
  "nodeCount": 1,
  "provider": "gcp",
  "region": "us-central1",
  "version": "1.23.8-gke.1900",
  "zone": "us-central1-c",
}
`;
